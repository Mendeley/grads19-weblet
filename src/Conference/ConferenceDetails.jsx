import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { getTimestring, getDatestring } from "../utils";
import { Link, useParams } from "react-router-dom";
import { getConferenceById } from "../api.js";

const StyledConferenceDetails = styled.div`
  height: 400px;
  padding: 20px;
`;

const StyledDetailsCardHeading = styled.h3`
  background: #dbd8db;
  width: 100%;
  height: 55px;
  font-size: 30px;
  font-weight: bold;
  text-align: center;
  vertical-align: middle;
`;

const StyledDetailsCard = styled.div`
  border-style: solid;
  border-color: black;
  border-radius: 25px;
  background: #dbd8db;
  width: 65%;
  height: 370px;
  transition: 0.3s;
  :hover {
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
  }
  vertical-align: middle;
  margin: 0 auto;
`;

export const StyledLink = styled(Link)`
  color: #7a517d;
`;

const ConferenceDetails = () => {
  const [isLoading, setIsLoading] = useState(true);
  const [conference, setConference] = useState(null);
  const [errorCaught, setErrorCaught] = useState(false);
  const { id } = useParams();

  const fetchData = async conferenceId => {
    setIsLoading(true);

    try {
      const conference = await getConferenceById(conferenceId);
      setConference(conference);
    } catch (error) {
      setErrorCaught(true);
    }
    setIsLoading(false);
  };

  useEffect(() => {
    if (id) {
      fetchData(id);
    }
  }, [id]);

  if (isLoading) {
    return <p>Loading...</p>;
  }

  if (errorCaught) {
    return <p>An error has occurred...</p>;
  }

  const { name, topic, dateTime, city, description } = conference || {};
  const date = new Date(dateTime);

  return (
    <StyledConferenceDetails>
      <StyledDetailsCard>
        <StyledDetailsCardHeading>{name}</StyledDetailsCardHeading>
        <p>{topic}</p>
        <p>{getDatestring(date)}</p>
        <p>{getTimestring(date)}</p>
        <p>{city}</p>
        <p>{description}</p>
        <StyledLink to={`/${id}/edit`}>Edit Conference</StyledLink>
      </StyledDetailsCard>
    </StyledConferenceDetails>
  );
};
export default ConferenceDetails;
