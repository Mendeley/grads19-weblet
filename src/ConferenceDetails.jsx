import React from "react";
// import Button from "./Button.jsx";
import styled from "styled-components";
import { getTimestring, getDatestring } from "./utils";

const StyledConferenceDetails = styled.div`
  width: 100%;
  height: 400px;
  color: white;
  background: darkslategray;
`;

const ConferenceDetails = ({ conference, setConferenceDetailsOpen }) => {
  const { name, topic, dateTime, city, description } = conference;
  const date = new Date(dateTime);

  return (
    <StyledConferenceDetails>
      <h3>{name}</h3>
      <p>{topic}</p>
      <p>{getDatestring(date)}</p>
      <p>{getTimestring(date)}</p>
      <p>{city}</p>
      <p>{description}</p>
      <button onClick={() => setConferenceDetailsOpen(false)}>Back</button>
    </StyledConferenceDetails>
  );
};
export default ConferenceDetails;
