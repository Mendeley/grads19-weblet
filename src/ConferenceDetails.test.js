import ConferenceDetails from "./ConferenceDetails";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

configure({ adapter: new Adapter() });

describe("ConferenceDetails", () => {
    const mockData = {
        id: 1,
        name: "Festival of Marketing",
        dateTime: "2019-11-12T12:34:11Z",
        city: "London",
        description:
            "From Festivalofmarketing.com: The Festival of Marketing is a unique experience where ambitious marketers can discover, learn, celebrate and shape the future together. As the largest global event dedicated to brand marketers, the Festival reflects the very nature of ...",
        topic: "Marketing"
    };
    it("renders conference details", () => {
        expect.assertions(1);

        const component = shallow(<ConferenceDetails conferences={mockData} />);

        expect(component.debug()).toMatchSnapshot();
    });
});
