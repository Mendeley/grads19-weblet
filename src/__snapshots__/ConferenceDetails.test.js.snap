// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConferenceDetails renders conference details 1`] = `
"<styled.div>
  <h3>
    Festival of Marketing
  </h3>
  <p>
    Marketing
  </p>
  <p>
    12/11/2019
  </p>
  <p>
    12:34
  </p>
  <p>
    London
  </p>
  <p>
    From Festivalofmarketing.com: The Festival of Marketing is a unique experience where ambitious marketers can discover, learn, celebrate and shape the future together. As the largest global event dedicated to brand marketers, the Festival reflects the very nature of ...
  </p>
  <button onClick={[Function: onClick]}>
    Back
  </button>
</styled.div>"
`;
